src/FMBot.Bot/Commands/Guild/UserGuildSettingCommands.cs

```diff
@@ -299,5 +299,71 @@ public async Task BlockedUsersAsync()
             await ReplyAsync("", false, this._embed.Build());
             this.Context.LogCommandUsed();
         }
+
+
+        [Command("whoknowswhitelist", RunMode = RunMode.Async)]
+        [Summary("Configures the WhoKnows charts to only show members with a specified role.\n\n" +
+                 "After changing your whitelist setting, you will most likely need to index the server again for it to take effect.\n\n" +
+                 "To remove the current whitelist setting, use this command without an option")]
+        [Examples("whoknowswhitelist", "whoknowswhitelist royals", "whoknowswhitelist 423946236102705154")]
+        [Alias("wkwhitelist", "wkwl")]
+        [GuildOnly]
+        public async Task SetWhoKnowsWhitelistRoleAsync([Remainder] string roleQuery = null)
+        {
+            var serverUser = (IGuildUser)this.Context.Message.Author;
+            if (!serverUser.GuildPermissions.BanMembers && !serverUser.GuildPermissions.Administrator &&
+                !await this._adminService.HasCommandAccessAsync(this.Context.User, UserType.Admin))
+            {
+                await ReplyAsync(
+                    "You are not authorized to use this command. Only users with the 'Ban Members' permission, server admins or FMBot admins can use this command.");
+                this.Context.LogCommandUsed(CommandResponse.NoPermission);
+                return;
+            }
+
+            if (string.IsNullOrEmpty(roleQuery) || roleQuery.ToLower() == "remove" || roleQuery.ToLower() == "delete")
+            {
+                await this._guildService.SetGuildWhoKnowsWhitelistRoleAsync(this.Context.Guild, null);
+                await this._guildService.StaleGuildLastIndexedAsync(this.Context.Guild);
+                await ReplyAsync("Removed the server's role whitelist for WhoKnows!");
+                this.Context.LogCommandUsed();
+                return;
+            }
+
+            if (
+                !(roleQuery.Length is >= 17 and <= 19) ||
+                !ulong.TryParse(roleQuery, out var discordRoleId)
+                )
+            {
+                var roleQueryResult = this.Context.Guild.Roles.First(r => r.Name.ToLower().Contains(roleQuery.ToLower()));
```

Comment:
`First` throws an exception if there is no result. Use `FirstOrDefault` instead

Reply:
Ah, when you first told me it wasn't "null safe" I imagined you meant it would return `null` rather than an empty interface

Reply:
Ah, my bad, should've been more clear. The `default` in `FirstOrDefault` is `null`. When using `First` there is no default value so it'll throw an exception when there are no results.

Reply:
Yea, I could have looked at the DataTip for that method to realize. Thank you for clearing it up for me